### PR TITLE
allow schema to optionally generate boxed types, closes #101

### LINKED ISSUE #101: Optionally allow builtin scalars to be generated with their wrapper types
This will be useful when certain request/response fields are allowed to be null. Since protocol buffers prohibits null values, using a wrapper type allows one to check of the value is set. This also allows the api to distinguish between default values and unset values.

See the following for details:

https://developers.google.com/protocol-buffers/docs/proto#optional

https://developers.google.com/protocol-buffers/docs/proto3#default

https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/wrappers.proto

### PR DESCRIPTION
**Pull request by drmaas**

### CODE REVIEW

**Comment by drmaas** [comment]
@jprobinson ptal

File: cmd/openapi2proto/main.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ func _main() error {
 	indent := flag.Int("indent", 4, "number of spaces used for indentation")
 	skipRpcs := flag.Bool("skip-rpcs", false, "skip rpc code generation. Defaults to false if not set")
 	namespaceEnums := flag.Bool("namespace-enums", false, "prefix enum values with the enum name to prevent namespace conflicts. Defaults to false if not set")
+	wrapPrimitives := flag.Bool("wrap-primitives", false, "specify non-required primitives values using their wrapper message types instead of their scalar types. Defaults to false if not set")
```
Comment (jprobinson) [inline cmd/openapi2proto/main.go]: Please update this to reflect the changes around required fields.

File: compiler/compiler.go [hunk 2]
```diff
@@ -657,7 +661,7 @@ func (c *compileCtx) compileSchema(name string, s *openapi.Schema) (protobuf.Typ
 	}
 }
 
-func (c *compileCtx) compileSchemaProperties(m *protobuf.Message, props map[string]*openapi.Schema) error {
+func (c *compileCtx) compileSchemaProperties(m *protobuf.Message, props map[string]*openapi.Schema, required []string) error {
```
Comment (jprobinson) [inline compiler/compiler.go]: We're not using the required fields anymore so we can drop the new param here.

File: proto_test.go [hunk 3]
```diff
@@ -37,6 +38,9 @@ func testGenProto(t *testing.T, tests ...genProtoTestCase) {
 			if test.options {
 				compilerOptions = append(compilerOptions, compiler.WithAnnotation(true))
 			}
+			if test.wrapNonRequired {
```
Comment (jprobinson) [inline proto_test.go]: Let's update this name to match the "wrap primitives" functionality.

**Comment by drmaas** [comment]
@jprobinson mind taking a last look?

**Comment by jprobinson** [comment]
🌮 ! thanks again!

**Pull request merged**